packages/esm-patient-notes-app/src/notes/ward-in-patient/notes-form.component.tsx

```diff
@@ -0,0 +1,166 @@
+
+type NotesFormData = z.infer<typeof noteFormSchema>;
+
+const noteFormSchema = z.object({
+  wardClinicalNote: z.string().refine((val) => val.trim().length > 0, {
+    message: translateFrom(moduleName, 'clinicalNoteErrorMessage', 'Clinical note is required'),
+  }),
+});
+
+const WardPatientNotesForm: React.FC<DefaultPatientWorkspaceProps> = ({
+  closeWorkspaceWithSavedChanges,
+  patientUuid,
+  promptBeforeClosing,
+}) => {
+  const { t } = useTranslation();
+  const session = useSession();
+  const config = useConfig<ConfigObject>();
+  const { clinicianEncounterRole, encounterNoteTextConceptUuid, encounterTypeUuid } = config.visitNoteConfig;
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [rows, setRows] = useState<number>();
+
+  const { control, handleSubmit, watch, getValues, setValue, formState } = useForm<NotesFormData>({
+    mode: 'onSubmit',
+    resolver: zodResolver(noteFormSchema),
+    defaultValues: {
+      wardClinicalNote: '',
+    },
+  });
+
+  const { isDirty } = formState;
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty, promptBeforeClosing]);
+
+  const locationUuid = session?.sessionLocation?.uuid;
+  const providerUuid = session?.currentProvider?.uuid;
+
+  const onSubmit = useCallback(
+    (data: NotesFormData) => {
+      const { wardClinicalNote } = data;
+      setIsSubmitting(true);
+
+      const notePayload = {
+        encounterDatetime: dayjs(new Date()).format(),
+        patient: patientUuid,
+        location: locationUuid,
+        encounterType: encounterTypeUuid,
+        encounterProviders: [
+          {
+            encounterRole: clinicianEncounterRole,
+            provider: providerUuid,
+          },
+        ],
+        obs: wardClinicalNote
+          ? [{ concept: { uuid: encounterNoteTextConceptUuid, display: '' }, value: wardClinicalNote }]
+          : [],
+      };
+
+      const abortController = new AbortController();
+
+      saveVisitNote(abortController, notePayload)
+        .then(() => {
+          closeWorkspaceWithSavedChanges();
+          showSnackbar({
+            isLowContrast: true,
+            subtitle: t('visitNoteNowVisible', 'It is now visible on the Visits page'),
+            kind: 'success',
+            title: t('visitNoteSaved', 'Visit note saved'),
+          });
+        })
+        .catch((err) => {
+          createErrorHandler();
+
+          showSnackbar({
+            title: t('visitNoteSaveError', 'Error saving visit note'),
+            kind: 'error',
+            isLowContrast: false,
+            subtitle: err?.message,
+          });
+        })
+        .finally(() => {
+          setIsSubmitting(false);
+        });
+    },
+    [
+      clinicianEncounterRole,
+      closeWorkspaceWithSavedChanges,
+      encounterNoteTextConceptUuid,
+      encounterTypeUuid,
+      locationUuid,
+      patientUuid,
+      providerUuid,
```

Comment:
You add these variables in the dependency array, but these values are in themselves not memoised, hence there's no advantage of using the `useCallback` here.

Always ensure that the variables used in the dependency array are themselves memoised.

Reply:
Errors i.e `Clinical note is required` are displayed on the form fields using the `invalid` and `invalidText` attributes